ci: failing build due to use of deprecated go/packages api

Build failed `staticcheck` linting due to:

https://travis-ci.org/mmcloughlin/avo/builds/542608355

```
build/context.go:38:9: SA1019: packages.LoadAllSyntax is deprecated: LoadAllSyntax exists for historical compatibility and should not be used. Please directly specify the needed fields using the Need values.  (staticcheck)
		Mode: packages.LoadAllSyntax,
		      ^
gotypes/signature_test.go:54:9: SA1019: packages.LoadTypes is deprecated: LoadTypes exists for historical compatibility and should not be used. Please directly specify the needed fields using the Need values.  (staticcheck)
		Mode: packages.LoadTypes,
		      ^
```

This refers to the following [`go/pacakges` documentation](https://godoc.org/golang.org/x/tools/go/packages):

```go
const (
    // Deprecated: LoadFiles exists for historical compatibility
    // and should not be used. Please directly specify the needed fields using the Need values.
    LoadFiles = NeedName | NeedFiles | NeedCompiledGoFiles

    // Deprecated: LoadImports exists for historical compatibility
    // and should not be used. Please directly specify the needed fields using the Need values.
    LoadImports = LoadFiles | NeedImports | NeedDeps

    // Deprecated: LoadTypes exists for historical compatibility
    // and should not be used. Please directly specify the needed fields using the Need values.
    LoadTypes = LoadImports | NeedTypes | NeedTypesSizes

    // Deprecated: LoadSyntax exists for historical compatibility
    // and should not be used. Please directly specify the needed fields using the Need values.
    LoadSyntax = LoadTypes | NeedSyntax | NeedTypesInfo

    // Deprecated: LoadAllSyntax exists for historical compatibility
    // and should not be used. Please directly specify the needed fields using the Need values.
    LoadAllSyntax = LoadSyntax
)
```

--- a/build/context.go
+++ b/build/context.go
@@ -35,7 +35,7 @@ func NewContext() *Context {
 // Package sets the package the generated file will belong to. Required to be able to reference types in the package.
 func (c *Context) Package(path string) {
 	cfg := &packages.Config{
-		Mode: packages.LoadAllSyntax,
+		Mode: packages.LoadSyntax,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

--- a/build/context.go
+++ b/build/context.go
@@ -35,7 +35,7 @@ func NewContext() *Context {
 // Package sets the package the generated file will belong to. Required to be able to reference types in the package.
 func (c *Context) Package(path string) {
 	cfg := &packages.Config{
-		Mode: packages.NeedName | packages.NeedFiles | packages.NeedCompiledGoFiles | packages.NeedImports | packages.NeedDeps | packages.NeedTypes | packages.NeedTypesSizes | packages.NeedSyntax | packages.NeedTypesInfo,
+		Mode: packages.NeedTypes | packages.NeedDeps,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

--- a/build/context.go
+++ b/build/context.go
@@ -35,7 +35,7 @@ func NewContext() *Context {
 // Package sets the package the generated file will belong to. Required to be able to reference types in the package.
 func (c *Context) Package(path string) {
 	cfg := &packages.Config{
-		Mode: packages.LoadSyntax,
+		Mode: packages.NeedName | packages.NeedFiles | packages.NeedCompiledGoFiles | packages.NeedImports | packages.NeedDeps | packages.NeedTypes | packages.NeedTypesSizes | packages.NeedSyntax | packages.NeedTypesInfo,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

--- a/gotypes/signature_test.go
+++ b/gotypes/signature_test.go
@@ -51,7 +51,7 @@ func TestLookupSignatureErrors(t *testing.T) {
 func LoadPackageTypes(t *testing.T, path string) *types.Package {
 	t.Helper()
 	cfg := &packages.Config{
-		Mode: packages.LoadTypes,
+		Mode: packages.LoadImports | packages.NeedTypes | packages.NeedTypesSizes,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

--- a/gotypes/signature_test.go
+++ b/gotypes/signature_test.go
@@ -51,7 +51,7 @@ func TestLookupSignatureErrors(t *testing.T) {
 func LoadPackageTypes(t *testing.T, path string) *types.Package {
 	t.Helper()
 	cfg := &packages.Config{
-		Mode: packages.NeedName | packages.NeedFiles | packages.NeedCompiledGoFiles | packages.NeedImports | packages.NeedDeps | packages.NeedTypes | packages.NeedTypesSizes,
+		Mode: packages.NeedTypes,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

--- a/gotypes/signature_test.go
+++ b/gotypes/signature_test.go
@@ -51,7 +51,7 @@ func TestLookupSignatureErrors(t *testing.T) {
 func LoadPackageTypes(t *testing.T, path string) *types.Package {
 	t.Helper()
 	cfg := &packages.Config{
-		Mode: packages.LoadImports | packages.NeedTypes | packages.NeedTypesSizes,
+		Mode: packages.NeedName | packages.NeedFiles | packages.NeedCompiledGoFiles | packages.NeedImports | packages.NeedDeps | packages.NeedTypes | packages.NeedTypesSizes,
 	}
 	pkgs, err := packages.Load(cfg, path)
 	if err != nil {

